### PR TITLE
Dropdown button: Add mainAction & update component

### PR DESCRIPTION
**Pull request by aprentout**
### What does this PR do?

Related to:[ #ENG-473](https://linear.app/upfluence/issue/ENG-473/make-buttondropdown-entirely-clickable)

### What are the observable changes?
Capture d’écran 2023-07-13 à 16 31 52

### Good PR checklist
- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [x] Added/updated documentation
- [ ] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: addon/components/o-s-s/button-dropdown.hbs
Comment (JulienVannier66) [inline addon/components/o-s-s/button-dropdown.hbs:1]: Some code duplication, but I don't really know how to factorize without added a lot of complexity and reduce readability. So LGTM; if @phndiaye or @Miexil have ideas? :smile:

File: tests/integration/components/o-s-s/button-dropdown-test.ts [hunk 1]
```diff
@@ -34,29 +35,120 @@ module('Integration | Component | o-s-s/button-dropdown', function (hooks) {
     assert.dom('.oss-button-dropdown__trigger .fx-row:first-child span').hasText('label');
   });
 
-  test('clicking on the caret container part opens the dropdown', async function (assert) {
-    await render(
-      hbs`<OSS::ButtonDropdown @icon="far fa-users" @label="label"><:items><div class="oss-button-dropdown__item">foo</div></:items></OSS::ButtonDropdown>`
-    );
+  module('If @mainAction is empty', function () {
```
Comment (JulienVannier66) [inline tests/integration/components/o-s-s/button-dropdown-test.ts:38]:
```suggestion
  module('If @mainAction is undefied', function () {
```

File: tests/integration/components/o-s-s/button-dropdown-test.ts [hunk 2]
```diff
@@ -34,29 +35,120 @@ module('Integration | Component | o-s-s/button-dropdown', function (hooks) {
...
+        .doesNotHaveStyle({ 'border-left': '1px solid rgb(229, 231, 235)' });
+    });
   });
 
-  test('it throws an error if no icon or label args is provided', async function (assert) {
-    setupOnerror((err: { message: string }) => {
-      assert.equal(
-        err.message,
-        'Assertion Failed: [component][OSS::ButtonDropdown] You must pass either a @label or an @icon argument.'
+  module('If @mainAction is set', function (hooks) {
```
Comment (JulienVannier66) [inline tests/integration/components/o-s-s/button-dropdown-test.ts:72]:
```suggestion
  module('If @mainAction is defined', function (hooks) {
```

File: tests/integration/components/o-s-s/button-dropdown-test.ts [hunk 3]
```diff
@@ -34,29 +35,120 @@ module('Integration | Component | o-s-s/button-dropdown', function (hooks) {
...
+      assert.dom('.oss-button-dropdown__trigger').hasNoAttribute('role', 'button');
+
+      await click('.oss-button-dropdown__trigger');
+      assert.dom('.oss-button-dropdown__items').doesNotExist();
     });
 
-    await render(
-      hbs`<OSS::ButtonDropdown><:items><div class="oss-button-dropdown__item">foo</div></:items></OSS::ButtonDropdown>`
-    );
+    test('Clicking on the right part should trigger mainAction', async function (assert) {
```
Comment (JulienVannier66) [inline tests/integration/components/o-s-s/button-dropdown-test.ts:89]:
```suggestion
    test('Clicking on the left part should trigger mainAction', async function (assert) {
```
Left part no?

File: addon/components/o-s-s/button-dropdown.hbs
Comment (aprentout) [reply addon/components/o-s-s/button-dropdown.hbs:1]: agree with that with a bunch of else if we can avoid duplication but loose readability

**Review by JulienVannier66 — APPROVED** [review]

**Pull request merged**